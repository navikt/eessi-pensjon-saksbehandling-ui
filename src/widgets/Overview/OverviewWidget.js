import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import ReactResizeDetector from 'react-resize-detector'
import Overview from './Overview'

const OverviewWidget = ({ onResize, onUpdate, widget }) => {
  const [mounted, setMounted] = useState(false)

  useEffect(() => {
    if (!mounted && onResize) {
      onResize()
      setMounted(true)
    }
  }, [mounted, onResize])

  return (
    <div className='w-OverviewWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      />
      <Overview onUpdate={onUpdate} widget={widget} />
    </div>
  )
}

OverviewWidget.properties = {
  type: 'overview',
  title: 'Overview widget',
  description: 'Widget with overview info',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {
    collapsed: false
  }
}

OverviewWidget.propTypes = {
  onResize: PT.func.isRequired,
  onUpdate: PT.func.isRequired,
  widget: PT.object.isRequired
}

export default OverviewWidget
