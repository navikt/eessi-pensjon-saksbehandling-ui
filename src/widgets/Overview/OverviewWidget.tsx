import { WidgetPropType } from 'declarations/Dashboard.pt'
import { WidgetFC, WidgetProps } from 'eessi-pensjon-ui/dist/declarations/Dashboard'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import ReactResizeDetector from 'react-resize-detector'
import Overview from './Overview'

const OverviewWidget: WidgetFC<WidgetProps> = ({
  onResize, onUpdate, widget
}: WidgetProps): JSX.Element => {
  const [mounted, setMounted] = useState<boolean>(false)

  useEffect(() => {
    if (!mounted && onResize) {
      onResize()
      setMounted(true)
    }
  }, [mounted, onResize])

  return (
    <div className='w-OverviewWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      />
      <Overview onUpdate={onUpdate} widget={widget} />
    </div>
  )
}

OverviewWidget.properties = {
  type: 'overview',
  title: 'Overview widget',
  description: 'Widget with overview info',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {
    collapsed: false
  }
}

OverviewWidget.propTypes = {
  onResize: PT.func.isRequired,
  onUpdate: PT.func.isRequired,
  widget: WidgetPropType.isRequired
}

export default OverviewWidget
