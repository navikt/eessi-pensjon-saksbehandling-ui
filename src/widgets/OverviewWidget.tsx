import { WidgetFC, WidgetProps } from '@navikt/dashboard'
import PT from 'prop-types'
import { useEffect } from 'react'
import ReactResizeDetector from 'react-resize-detector'
import Overview from './Overview/Overview'

const OverviewWidget: WidgetFC<WidgetProps> = ({
  onResize, onUpdate, widget
}: WidgetProps): JSX.Element => {

  useEffect(() => {
    onResize()
  }, [])

  return (
    <div data-testid='w-OverviewWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      >
        <Overview onUpdate={onUpdate} widget={widget} />
      </ReactResizeDetector>
    </div>
  )
}

OverviewWidget.properties = {
  type: 'overview',
  title: 'Overview widget',
  description: 'Widget with overview info',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {
    collapsed: false
  }
}

OverviewWidget.propTypes = {
  onResize: PT.func.isRequired,
  onUpdate: PT.func.isRequired,
  widget: PT.any.isRequired // WidgetPropType.isRequired
}

export default OverviewWidget
