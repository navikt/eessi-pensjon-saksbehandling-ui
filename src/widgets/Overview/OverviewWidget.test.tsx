import { WidgetProps } from 'nav-dashboard'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import OverviewWidget from './OverviewWidget'

jest.mock('widgets/Overview/Overview', () => () => (<div className='mock-w-overview' />))

describe('widgets/OverviewWidget', () => {
  let wrapper: ReactWrapper
  const initialMockProps: WidgetProps = {
    onResize: jest.fn(),
    onFullFocus: jest.fn(),
    onRestoreFocus: jest.fn(),
    onUpdate: jest.fn(),
    widget: {
      i: 'i',
      type: 'overview',
      visible: true,
      title: 'Overview',
      options: {
        collapsed: false
      }
    }
  }

  beforeEach(() => {
    wrapper = mount(<OverviewWidget {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('UseEffect: it tries to resize', () => {
    expect(initialMockProps.onResize).toHaveBeenCalled()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.w-OverviewWidget')).toBeTruthy()
    expect(wrapper.find('.mock-w-overview')).toBeTruthy()
  })

  it('Has properties', () => {
    expect(OverviewWidget.properties).toHaveProperty('type')
    expect(OverviewWidget.properties).toHaveProperty('title')
    expect(OverviewWidget.properties).toHaveProperty('description')
    expect(OverviewWidget.properties).toHaveProperty('layout')
    expect(OverviewWidget.properties).toHaveProperty('options')
  })
})
