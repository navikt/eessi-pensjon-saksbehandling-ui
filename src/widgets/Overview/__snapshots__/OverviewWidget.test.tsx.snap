// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widgets/OverviewWidget Renders 1`] = `
<OverviewWidget
  onFullFocus={[MockFunction]}
  onResize={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onRestoreFocus={[MockFunction]}
  onUpdate={[MockFunction]}
  widget={
    Object {
      "i": "i",
      "options": Object {
        "collapsed": false,
      },
      "title": "Overview",
      "type": "overview",
      "visible": true,
    }
  }
>
  <div
    className="w-OverviewWidget"
  >
    <ResizeDetector
      handleHeight={true}
      handleWidth={true}
      nodeType="div"
      onResize={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      querySelector={null}
      refreshRate={1000}
      skipOnMount={false}
      targetDomEl={null}
      targetRef={null}
    >
      <div />
    </ResizeDetector>
    <Component
      onUpdate={[MockFunction]}
      widget={
        Object {
          "i": "i",
          "options": Object {
            "collapsed": false,
          },
          "title": "Overview",
          "type": "overview",
          "visible": true,
        }
      }
    >
      <div
        className="mock-w-overview"
      />
    </Component>
  </div>
</OverviewWidget>
`;
